contracts/BookMarketplace.sol

```diff
@@ -44,15 +48,98 @@ contract BookMarketplace is ERC721URIStorage {
         bool sold
     );
 
+    event ListingUpdated(
+        uint256 indexed tokenId,
+        address seller,
+        address owner,
+        uint256 instantPrice,
+        uint256 startingPrice,
+        bool allowBid,
+        Bid[] bidList,
+        bool sold
+    );
+
+    /* modifiers */
+    
+    
+    // Modifier prevents function from being called by anyone other than the market owner
+    modifier onlyMarketOwner() {
+        require(
+            marketOwner == msg.sender,
+            "Only the marketplace owner can call this function."
+        );
+        _;
+    }
+
+    /* constructor */
+
     constructor() ERC721("BookExchange Tokens", "BOOK") {
-        owner = payable(msg.sender);
+        marketOwner = payable(msg.sender);
+    }
+
+    /* functions */
+
+    function getMarketOwnerAddress() public view returns(address) {
+        return marketOwner;
+    }
+
+    // change the price of a listing
+    function updateInstantPrice(uint256 tokenId, uint256 newPrice) public payable onlyMarketOwner {
```

Comment:
payable should only be used when we need to transfer fees. In this case is it necessary?

Reply:
Ah, this was a mistake I think, will remove and update.

Reply:
I think you forgot to remove the payable here.